src/components/importkml/ImportKmlDirective.js

```diff
@@ -51,24 +51,27 @@ goog.require('ga_urlutils_service');
             $scope.canceler = $q.defer();
 
             // Angularjs doesn't handle onprogress event
-            $http.get(gaUrlUtils.proxifyUrl($scope.fileUrl), {
-              cache: true,
-              timeout: $scope.canceler.promise
-            }).then(function(response) {
-              var data = response.data;
-              var fileSize = response.headers('content-length');
-              if (gaKml.isValidFileContent(data) &&
-                  gaKml.isValidFileSize(fileSize)) {
-                $scope.userMessage = $translate.instant('upload_succeeded');
-                $scope.fileContent = data;
-                $scope.fileSize = fileSize;
-              } else {
+            gaUrlUtils.proxifyUrl($scope.fileUrl).then(function(opts) {
+              $http.get(opts.url, {
+                headers: opts.headers,
```

Comment:
it seems the headers are always the same here. Why not putting it directly here instead of opts obj.